src/pocketmine/event/block/SignChangeEvent.php

```diff
@@ -69,14 +69,30 @@ public function getLines(){
 	 * @return string
 	 */
 	public function getLine($index){
+		if($index < 0 or $index > 3){
+			throw new \InvalidArgumentException("Index must be in the range 0-3!");
+		}
 		return $this->lines[$index];
 	}
 
 	/**
+	 * @param string[] $lines
+	 */
+	public function setLines(array $lines){
+		if(count($lines) > 4){
```

Comment:
@robske110 That would depend on #1204. It is not guaranteed that SignChangeEvent is always for an empty sign; plugins may fire it for things like editing signs.